--- frameworks/base/services/core/java/com/android/server/connectivity/Tethering.java.orig	2018-03-26 11:03:41.607870948 +0800
+++ frameworks/base/services/core/java/com/android/server/connectivity/Tethering.java	2018-03-26 11:04:20.295512856 +0800
@@ -976,6 +976,12 @@
             return;
         }
 
+        // Hack for fix WIFI Tether on some devices
+        if(mTetherStates.get(chosenIface) == null) {
+            Log.e(TAG, "Adding non-existent Iface: " + chosenIface);
+            maybeTrackNewInterfaceLocked(chosenIface, interfaceType);
+        }
+
         changeInterfaceState(chosenIface, requestedState);
     }
 
